lib/curly/configuration.rb

```diff
@@ -0,0 +1,25 @@
+module Curly
+  class << self
```

Comment:
Since you are only defining methods on `self`, you could use the `extend self` pattern here:
```rb
module Curly
  extend self

  def configuration
    @configuration ||= Configuration.new
  end

  def configuration=(c)
    @configuration = c
  end

  def configure
    yield(configuration)
  end

  def reset
    @configuration = Configuration.new
  end

  class Configuration
    attr_accessor :cache_store

    def initialize
      @cache_store = nil
    end
  end
end
```